### PR TITLE
feat: steps support for circular progress bar

### PR DESCRIPTION
**Pull request by yykoypj**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/35802

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |steps support for circular progress bar           |
| 🇨🇳 Chinese |Progress: circle模式下支持steps           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 85b58aa</samp>

This pull request adds a new feature to the `Progress` component, which allows rendering a circular progress bar with steps. It modifies the `steps` prop, updates the documentation and the styles, and adds a demo for the new feature.

### <samp>🤖 Generated by Copilot at 85b58aa</samp>

*  Add the `steps` prop to the `Progress` component, which allows the user to specify the total step count for the circular or linear progress bar. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaR32), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL54-R54))
*  Modify the `Progress` value or function to render the `Progress` component based on the `steps` prop. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL134-R139), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL164-R171))
*  Modify the `Circle` component to render the `RCCircle` component from the `rc-progress` library with the `steps` prop and the `percentArray` variable, which contains an array of percentages for the progress bar. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaR55), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaL66-R72))
*  Add a demo for the circular progress bar with steps in the `components/progress/demo` folder, which renders two examples of the circular progress bar with steps, one with the `dashboard` type and one with the `circle` type. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-be09355126aad10e535669fde2deb475cbdccec0bab2789f53ccd2cd2c2241f6R1-R9), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-11dc9a368d3395b2b72019300e7837ff1d2553e16f1994419704feca9c9420f7R1-R26))
*  Add a code block to the `Examples` section of the English and Chinese documentation for the `Progress` component, which references the demo file for the circular progress bar with steps. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bR36), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffR37))
*  Add a row to the `API` table of the English and Chinese documentation for the `Progress` component, which describes the `steps` prop for the `Progress` component. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bR55), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffR56))
*  Remove a row from the `type="line"` subsection of the `API` table of the English and Chinese documentation for the `Progress` component, which describes the `steps` prop for the `Progress` component, but is redundant since the `steps` prop is now applicable to all types of the `Progress` component. ([link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL59), [link](https://github.com/ant-design/ant-design/pull/42815/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL60))

### CODE REVIEW

File: components/progress/index.en-US.md [hunk 1]
```diff
@@ -51,12 +52,12 @@ Properties that shared by all types.
 | trailColor | The color of unfilled part | string | - | - |
 | type | To set the type, options: `line` `circle` `dashboard` | string | `line` |
 | size | Progress size | number \| \[number, number] \| "small" \| "default" | "default" | v5.3.0 |
+| steps | The total step count | number \| { count: number, space: number } | - | object: 5.5.3 |
 
 ### `type="line"`
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
-| steps | The total step count | number | - | - |
```
Comment (kiner-tang) [inline components/progress/index.en-US.md:59]: 这里为什么要删除，不是只改了进度圈吗？这个是进度条的文档

**Review by kiner-tang — COMMENTED** [review]
新增新特性提到 feature 分支吧。

**Comment by kiner-tang** [comment]
加个测试用例呗

**Comment by yykoypj** [comment]
new pull request: https://github.com/ant-design/ant-design/pull/42821

**Pull request closed without merging**